pageserver/src/bin/pageserver.rs

```diff
@@ -465,14 +472,25 @@ fn start_pageserver(conf: &'static PageServerConf) -> Result<()> {
 
     // Spawn a thread to listen for connections. It will spawn further threads
     // for each connection.
+    let term_pageservice = Arc::clone(&term);
     let page_service_thread = thread::Builder::new()
         .name("Page Service thread".into())
         .spawn(move || {
-            page_service::thread_main(conf, auth, pageserver_listener, conf.auth_type)
+            page_service::thread_main(
+                term_pageservice,
+                conf,
+                auth,
+                pageserver_listener,
+                conf.auth_type,
+            )
         })?;
 
     join_handles.push(page_service_thread);
 
+    while !term.load(Ordering::Relaxed) {}
```

Comment:
This will busy-wait. I think you want to follow [this example](https://docs.rs/signal-hook/0.3.10/signal_hook/iterator/index.html#examples) from the 'signal_hook' docs instead